src/lib/comms/sol-coap.c

```diff
@@ -1503,19 +1512,23 @@ respond_packet(struct sol_coap_server *server, struct sol_coap_packet *req,
     /* If it isn't a request. */
     if (code & ~SOL_COAP_REQUEST_MASK) {
         bool found_reply = false;
+        bool reply_callback_result = false;
         SOL_PTR_VECTOR_FOREACH_REVERSE_IDX (&server->pending, reply, i) {
             if (!match_reply(reply, req))
                 continue;
 
-            if (!reply->cb((void *)reply->data, server, req, cliaddr)) {
-                sol_ptr_vector_del(&server->pending, i);
```

Comment:
Is it important that the reply is deleted from the vector *before* the unobserve packet is sent? I ask, because this modification would call ```send_unobserve_packet()``` before removing the pending reply from the vector in case an observation is in place.